pkg/controller/cronjob/utils.go

```diff
@@ -204,6 +206,13 @@ func getJobFromTemplate2(cj *batchv1.CronJob, scheduledTime time.Time) (*batchv1
 	return job, nil
 }
 
+// appendJobCreationTime returns job annotations after appending job creation timestamp.
+func appendJobCreationTimestamp(creationTimestamp time.Time, tAnnotations labels.Set) labels.Set {
+	nameSet := make(labels.Set, 1)
+	nameSet[batchv1.JobCreationTimestampAnnotation] = creationTimestamp.String()
```

Comment:
A few comments here:
1. I don't think we need entire new function just for adding one more annotation. I'd leave this line
```golang
annotations := copyAnnotations(&cj.Spec.JobTemplate)
```
which ensures the set exists, and then just:
```golang
annotations[batchv1.JobCreationTimestampAnnotation] = <time>
```
2. `creationTimestamp.String()` isn't safe since it may change, and since we expose this information as API we need to ensure the stability of the data. I'd suggest having an explicit date-time format: `YYYY-MM-DD HH:MM`, for example. You also need to be explicit about what timezone this value is in, since we also support that, so it's reasonable to think through which one we want to present here.

Reply:
Both comments have been addressed.